shared/src/commonMain/kotlin/io/customer/shared/tracking/api/TrackingHttpClient.kt

```diff
@@ -0,0 +1,61 @@
+package io.customer.shared.tracking.api
+
+import io.customer.shared.tracking.api.model.BatchTrackingRequestBody
+import io.customer.shared.tracking.api.model.BatchTrackingResponse
+import io.customer.shared.tracking.api.model.BatchTrackingResponseBody
+import io.customer.shared.tracking.api.model.toTrackingRequest
+import io.customer.shared.tracking.model.Task
+import io.customer.shared.util.Logger
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+
+/**
+ * HTTP client interface responsible for making all network calls in the SDK.
+ *
+ * The class doesn't perform any validation and is only responsible for transforming and passing
+ * filtered data to the server.
+ */
+internal interface TrackingHttpClient {
+    /**
+     * Dispatch all tasks passed to the server in a single batch call.
+     */
+    suspend fun track(tasks: List<Task>): Result<BatchTrackingResponse>
+}
+
+internal class TrackingHttpClientImpl(
+    private val logger: Logger,
+    private val httpClient: HttpClient,
+) : TrackingHttpClient {
+    override suspend fun track(tasks: List<Task>): Result<BatchTrackingResponse> {
+        val result = kotlin.runCatching {
```

Comment:
Do we need `kotlin.runCatching {}` here? Since this is a `suspend` function, I don't see the need for `runCatching`